webanno-api-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/annotation/event/RelationUpdateEvent.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019
+ * Ubiquitous Knowledge Processing (UKP) Lab and FG Language Technology
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.clarin.webanno.api.annotation.event;
+
+import org.apache.uima.cas.Feature;
+import org.apache.uima.cas.text.AnnotationFS;
+import org.springframework.context.ApplicationEvent;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.adapter.RelationAdapter;
+import de.tudarmstadt.ukp.clarin.webanno.model.SourceDocument;
+
+public class RelationUpdateEvent extends ApplicationEvent
+{
+    private static final long serialVersionUID = -8621413642390759892L;
+    
+    private final SourceDocument document;
+    private final String user;
+    private final AnnotationFS annotation;
+
+    public RelationUpdateEvent(Object aSource, SourceDocument aDocument, String aUser,
+            AnnotationFS aAnnotation)
+    {
+        super(aSource);
+        document = aDocument;
+        user = aUser;
+        annotation = aAnnotation;
+    }
+
+    public SourceDocument getDocument()
+    {
+        return document;
+    }
+
+    public String getUser()
+    {
+        return user;
+    }
+
+    public AnnotationFS getAnnotation()
+    {
+        return annotation;
+    }
+    
+    @Override
+    public String toString()
+    {
+        String govFeatureName = ((RelationAdapter) getSource()).getSourceFeatureName();
```

Comment:
I think we should avoid casting the source to RelationAdapter here because there is no guarantee that the source actually is a RelationAdapter (cf. source being an Object in the constructor). An option might be to explicitly pass the relation adapter as an additional type-checked argument. I'd also suggest considering not to pass the entire adapter around but extract from it what information is relevant. Mind that the `toString()` method should only have meaning for debugging and in other cases shouldn't really be used. Yet another option might be to check if the source is actually an instance of RelationAdapter and if yes use it for the toString method... but it would seem a bit hacky because in some situation toString() would produce one format and in others another format.

Reply:
Thanks for the suggestion. I implemented the requested changes by also passing the source annotation to the vent.